code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,41 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
```

Comment:
Ah, so the model is named after how it integrates with the view, not with how it integrates with the Controller/Action.  I like that convention now that I understand it.

Lets leave the class where it is, but change the name to `IndexViewModel`, since this is the view model for the entire Index page.  It happens to only contain details about the meta tags at the moment, but over time it may grow to include other parts of the index page and I don't think you can have multiple models per View?